### ISSUE TITLE
[JENKINS-3891] Build 'appears to be stuck' even if in queue

### ISSUE THREAD

**Issue by timja** [timja/jenkins-gh-issues-poc-06-20#24]
<p>If I understand the abort stuck builds feature, it is intended to kill builds<br/>
that hang due to some sort of execution.  However, builds are reported to be<br/>
stuck even if they are simply waiting their turn in the queue.  It seems like<br/>
this timer should not start until the job begins execution.</p>

---
<details><summary><i>Originally reported by <a title="John McNair" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=pamdirac">pamdirac</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-3891" target="_blank">Build 'appears to be stuck' even if in queue</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Kohsuke Kawaguchi" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2009-06-23T15:36:37+00:00
<li><b>imported</b>: 2022/01/10
</ul></i>
</details>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>The intention of marking such jobs in the queue is to let you know that these<br/>
are starving for additional executors &#8212; it's time to add more slaves.</p>

<p>Perhaps we should change the wording a bit? Like from "stuck" to "starving<br/>
executors" or something like that? With perhaps a hyperlink to a page in Wiki<br/>
that talks about what you are supposed to do when you see it?</p>

<p>Or maybe we should change the color from red to something less scary, like yellow?</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>:</i>
<p>when you mouseover the link I think it already shows the reason (awaiting an<br/>
executor...).. maybe change the text to an icon and keep the mouseover text. <br/>
confusing wording gone <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=pamdirac">pamdirac</a>:</i>
<p>Ah, so I think I misunderstood the feature.  When I set a value for "Abort the<br/>
build if it's stuck", this same timeout applies separately to executor<br/>
starvation and the actual build execution time?</p>

<p>I guess what's weird is that we have different jobs that take wildly varying<br/>
lengths of time.  So I have timeouts for various jobs at 5 minutes, 30 minutes,<br/>
1 hour and 3 hours.  For a couple of hours during the day for us, it's very easy<br/>
to get a 15-minute queue depth.  The five minute jobs will then be reported as<br/>
stuck while sitting in the queue, but the other jobs won't.</p>

<p>I think that just a better indication of what is really happening is an ok solution.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in hudson<br/>
User: : jglick<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/resources/lib/hudson/queue.jelly<br/>
 trunk/hudson/main/war/images/hourglass.svg<br/>
 trunk/hudson/main/war/resources/images/16x16/hourglass.gif<br/>
 trunk/hudson/main/war/resources/images/24x24/hourglass.gif<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19163" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19163</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-3891&#93;</span> Replaced misleading, verbose, and scary red "(appears to be stuck)" label with a warning icon & tooltip.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in hudson<br/>
User: : jglick<br/>
Path:<br/>
 trunk/www/changelog.html<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19164" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19164</a><br/>
Log:<br/>
<a href="https://issues.jenkins.io/browse/JENKINS-3891" title="Build &#39;appears to be stuck&#39; even if in queue" class="issue-link" data-issue-key="JENKINS-3891"><del>JENKINS-3891</del></a> Noting.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3040 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by timja** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja/jenkins-gh-issues-poc-06-20/issues?q=in%3Atitle%20JENKINS-3040">JENKINS-3040</a>]</i>
